plugins/code-coverage/src/components/Router.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { MissingAnnotationEmptyState } from '@backstage/core';
+import { CodeCoveragePage } from './CodeCoveragePage';
+
+export const isCodeCoverageAvailable = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.['backstage.io/code-coverage']);
+
+export const Router = () => {
+  const { entity } = useEntity();
+
+  if (!isCodeCoverageAvailable(entity)) {
+    return (
+      <MissingAnnotationEmptyState annotation="backstage.io/code-coverage" />
+    );
+  }
+  return <CodeCoveragePage />;
```

Comment:
@Rugvip what's the stance on composability? accept and fix later, or make sure to only get new-style code in going forward? Would you mind reviewing just this file and the few ones below in that regard?